rosidl_gen/idl_generator.js

```diff
@@ -42,7 +43,7 @@ async function writeGeneratedCode(dir, fileName, code) {
   await fse.writeFile(path.join(dir, fileName), code);
 }
 
-function generateServiceJSStruct(serviceInfo, dir) {
+async function generateServiceJSStruct(serviceInfo, dir) {
```

Comment:
@wayneparrott after looking through [rep-2012.rst](https://github.com/ros-infrastructure/rep/blob/336d3b66135ca71a82d5e903de1e865e4ceddd2d/rep-2012.rst), now I understand what is the Service Introspection better.

I am thinking if we can leverage the `dot` to generate the `.msg` file, like `AddTwoInts_Event.msg`
```
rcl_interfaces/msg/ServiceEventInfo info

example_interfaces/srv/AddTwoInts_Request[<=1] request

example_interfaces/srv/AddTwoInts_Response[<=1] response
```

I tried with the following `service_event.dot`:
```
rcl_interfaces/msg/ServiceEventInfo info

{{=it.serviceInfo.pkgName}}/{{=it.serviceInfo.subFolder}}/{{=it.serviceInfo.interfaceName}}_Request[<=1] request;

{{=it.serviceInfo.pkgName}}/{{=it.serviceInfo.subFolder}}/{{=it.serviceInfo.interfaceName}}_Response[<=1] response;
```
and hack the method `generateServiceJSStruct`
```js
function generateServiceJSStruct(serviceInfo, dir) {
  dir = path.join(dir, `${serviceInfo.pkgName}`);
    const fileName = serviceInfo.interfaceName + '_Event.msg';
    const generatedEvent = removeEmptyLines(
      dots.service_event({ serviceInfo: serviceInfo })
    );
    return writeGeneratedCode(dir, fileName, generatedEvent);
}
```
and corresponding generated `AddTwoInts_Event.msg` file is
```
rcl_interfaces/msg/ServiceEventInfo info
example_interfaces/srv/AddTwoInts_Request[<=1] request;
example_interfaces/srv/AddTwoInts_Response[<=1] response;
```
The final step is using the generated `.msg` file to create `.js` file, which as this PR does, how do you think? I'm fine we can land this implementation and see if we could simplify it later, thanks!